### PR TITLE
Edit ACMG classifications

### LINKED ISSUE #4208: Add an Edit-function for authors of ACMG classifications in Scout
Can an Edit-function be added to the ACMG-classification-module in Scout, available to the author of each ACMG-classification? For example, when we have meetings about a variant, with the author of a classification presenting, we would like for the author to be able to edit the ACMG-classification without having to recreate it from scratch. I believe this would also save time for everybody involved in analysis as any minor error means they have to fill in the form "Evidence of pathogenicity" all over.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Add a link to edit the ACMG classification of a variant (only if classiciation has criteria, i.e. made via classification page) - fix #4208

## ----> The edit action will create a new classification! <----

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/variant/templates/variant/acmg.html [hunk 1]
```diff
@@ -29,22 +29,27 @@
     <form action="{{ url_for('variant.variant_acmg', institute_id=institute._id, case_name=case.display_name, variant_id=variant._id) }}" method="POST">
       <div class="card panel-default mt-3">
         <div class="card-body">
-          {% if not evaluation %}
-              <div class="mt-3 fixed-bottom bg-light border">
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/acmg.html:33]: Moved down

File: scout/server/blueprints/variant/templates/variant/acmg.html [hunk 2]
```diff
@@ -29,22 +29,27 @@
...
-            <h4>
+          {% if evaluation %}
+             <h4>
               {{ evaluation.classification.label }}
               <span class="badge bg-info">{{ evaluation.classification.short }}</span>
             </h4>
             By {{ evaluation.user_name }} on {{ evaluation.created_at.date() }}
+            {% if edit %}
+              <br><br>
+              <button class="btn btn-primary form-control">Update</button>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/acmg.html:40]: I really like that we give the users the feature to continue based off a given evaluation, while still preserving the audit log. While edit is a useful short name, this is not really an edit, but rather a cloning or re-classification. And in particular clicking update doesn't really update, rather create a new event. How about just renaming this button to indicate that; say "reclassify", "resubmit", or "add"?
```suggestion
              <button class="btn btn-primary form-control">Reclassify</button>
```

File: scout/server/blueprints/variant/templates/variant/acmg.html [hunk 3]
```diff
@@ -120,6 +125,11 @@ <h5>Likely Pathogenic</h5>
   {{ super() }}
 
   <script>
+
+    window.onload=function() {
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/acmg.html:129]: 👍

**Review by dnil — APPROVED** [review]
Great! Simple change, yet solves both the users issue and preserves audit log. Approving since it works just fine, but consider renaming at least the "update" button, and possibly some info about the feature, e.g. a  tool-tip to the same button, and perhaps also the `Edit` link.

File: scout/server/blueprints/variant/templates/variant/acmg.html (hunk 2 above)
Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/acmg.html]: Good suggestion, applied!

Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/acmg.html]: Added tooltips as well:

image

image

**Pull request merged**